blurring_as_a_service/inference_pipeline/components/detect_and_blur_sensitive_data.py

```diff
@@ -53,43 +54,56 @@ def detect_and_blur_sensitive_data(
+    # Iterate over files in the folder
+    for batch_file_txt in os.listdir(batches_files_path):
+        file_path = os.path.join(batches_files_path, batch_file_txt)
+        # Check if the path points to a file (not a directory)
+        if os.path.isfile(file_path):
+            print(f"Creating inference step: {file_path}")
 
-    data = dict(
-        train=f"../{files_to_blur_full_path}",
-        val=f"../{files_to_blur_full_path}",
-        test=f"../{files_to_blur_full_path}",
-        nc=2,
-        names=["person", "license_plate"],
-    )
+            files_to_blur_full_path = os.path.join(
+                "outputs", batch_file_txt
+            )  # use outputs folder as Azure expects outputs there
+            with open(file_path, "r") as src:
+                with open(files_to_blur_full_path, "w") as dest:
+                    for line in src:
+                        dest.write(f"{mounted_root_folder}/{line}")
+                        print(f"{mounted_root_folder}/{line}")
 
-    # TODO create postgresql string and send to val.py
+            data = dict(
+                train=f"../{files_to_blur_full_path}",
+                val=f"../{files_to_blur_full_path}",
+                test=f"../{files_to_blur_full_path}",
+                nc=2,
+                names=["person", "license_plate"],
+            )
 
-    with open(f"{yolo_yaml_path}/pano.yaml", "w") as outfile:
-        yaml.dump(data, outfile, default_flow_style=False)
-    cuda_device = torch.cuda.current_device()
-    model_parameters = json.loads(model_parameters_json)
-    val.run(
-        weights=f"{mounted_root_folder}/best.pt",  # TODO get from Azure ML models
-        data=f"{yolo_yaml_path}/pano.yaml",
-        project=results_path,
-        device=cuda_device,
-        name="",
-        customer_name=customer_name,  # We want to save this info in a database
-        **model_parameters,
-    )
+            # TODO create postgresql string and send to val.py
 
-    try:
-        os.remove(batch_file_txt)
-    except OSError as error:
-        raise OSError(f"Failed to remove file '{batch_file_txt}': {error}")
+            with open(f"{yolo_yaml_path}/pano.yaml", "w") as outfile:
+                yaml.dump(data, outfile, default_flow_style=False)
+            cuda_device = torch.cuda.current_device()
+            model_parameters = json.loads(model_parameters_json)
+            database_parameters = json.loads(database_parameters_json)
+            val.run(
+                weights=f"{mounted_root_folder}/best.pt",  # TODO get from Azure ML models
+                data=f"{yolo_yaml_path}/pano.yaml",
+                project=results_path,
+                device=cuda_device,
+                name="",
+                customer_name=customer_name,  # We want to save this info in a database
+                **model_parameters,
+                **database_parameters,
+            )
+
+            try:
```

Comment:
If there is only one file left, also the folder will be deleted, is this a problem?

Reply:
No, not a problem